FactoryWeather/App/VIP Search/DataSource/SearchDataSource.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  SearchDataSource.swift
+//  FactoryWeather
+//
+//  Created by Matej Korman on 14/01/2019.
+//  Copyright © 2019 Matej Korman. All rights reserved.
+//
+
+import Foundation
+
+class SearchDataSource: DataSourceProtocol {
+    var sections = [SearchSection]()
+    private var locations = [Location]()
+    private var filterText: String?
+    
+    func addLocations(_ locations: [Location]) {
+        locations.reversed().forEach { location in
+            if !self.locations.contains(where: { $0 == location }) {
+                self.locations.insert(location, at: 0)
+            }
+        }
+        buildSections(forText: filterText)
+    }
+    
+    func filterLocations(forText text: String) {
+        filterText = text
+        buildSections(forText: text)
+    }
+}
+
+private extension SearchDataSource {
+    func buildSections(forText text: String?) {
+        sections.removeAll()
+        var rows = [SearchRow]()
+        if let text = text,
+            !text.isEmpty {
```

Comment:
What do you think about turning

```
if let text = text,
	            !text.isEmpty
```

into 

`if let text = text, !text.isEmpty`

Much more readable.

Reply:
It's a lot of `text`s in one row, but I'll change that. I got used to this way of writing statements so it's more readable to me.

Reply:
I don't mind if you leave it as is. 

My personal preference is that I find short `if`'s like this more readable as a one-liner 🙂